tests/test_solver_based_model.py

```diff
@@ -958,6 +958,21 @@ def test_essential_reactions(self):
                 self.model.reactions.Biomass_Ecoli_core_N_LPAREN_w_FSLASH_GAM_RPAREN__Nmet2.lower_bound = 999999.
                 self.model.essential_reactions()
 
+        def test_essential_metabolites(self):
+            model = self.model.copy()
+            essential_metabolites_unbalanced = [m.id for m in model.essential_metabolites(force_steady_state=False)]
+            essential_metabolites_balanced = [m.id for m in model.essential_metabolites(force_steady_state=True)]
+            self.assertTrue(sorted(essential_metabolites_unbalanced) == sorted(ESSENTIAL_METABOLITES))
+            self.assertTrue(sorted(essential_metabolites_balanced) == sorted(ESSENTIAL_METABOLITES))
+
+            with self.assertRaises(cameo.exceptions.SolveError):
+                self.model.reactions.Biomass_Ecoli_core_N_LPAREN_w_FSLASH_GAM_RPAREN__Nmet2.lower_bound = 999999.
+                self.model.essential_metabolites(force_steady_state=False)
+
+            with self.assertRaises(cameo.exceptions.SolveError):
```

Comment:
This one craps out with GLPK. It seems to be a(nother) weird issue where GLPK gets stuck with an invalid basis, which results in an undefined solution. It can't get out of this without some help. I think I've fixed it in optlang.